docs/ispc.rst

```diff
@@ -1587,40 +1592,59 @@ SI-inspired suffixes "k", "M", and "G" respectively:
    int two_megs = 2M; // 2 * 1024 * 1024
    int one_gig = 1G;  // 1024 * 1024 * 1024
 
-Floating-point constants can be specified in one of three ways.  First,
-they may be a sequence of zero or more digits from 0 to 9, followed by a
-period, followed by zero or more digits from 0 to 9. (There must be at
-least one digit before or after the period).
+Floating Point Literals
+-----------------------
 
-The second option is scientific notation, where a base value is specified
-as the first form of a floating-point constant but is then followed by an
-"e" or "E", then a plus sign or a minus sign, and then an exponent.
+ISPC supports 3 floating point types : ``float16``, ``float`` and ``double``.
 
-Finally, floating-point constants may be specified as hexadecimal
-constants; this form can ensure a perfectly bit-accurate representation of
-a particular floating-point number.  These are specified with an "0x"
-prefix, followed by a zero or a one, a period, and then the remainder of
-the mantissa in hexadecimal form, with digits from 0-9, a-f, or A-F.  The
-start of the exponent is denoted by a "p", which is then followed by an
-optional plus or minus sign and then digits from 0 to 9.  For example:
+* ``float16`` is a half-precision(16 bit format) floating point type.
+* ``float`` is a single-precision(32 bit format) floating point type.
+* ``double`` is a double-precision(64 bit format) floating point type.
 
-::
+All 3 floating-point constants can be specified in one of three ways.
+
+* First, they may be a sequence of zero or more digits from 0 to 9, followed
+  by a period, followed by zero or more digits from 0 to 9. (There must be at
+  least one digit before or after the period).
+* The second option is scientific notation, where a base value is specified
+  as the first form of a floating-point constant but is then followed by an
+  "e" or "E", then a plus sign or a minus sign, and then an exponent.
+* Finally, floating-point constants may be specified as hexadecimal.
+  constants; this form can ensure a perfectly bit-accurate representation of
+  a particular floating-point number.  These are specified with an "0x"
+  prefix, followed by a zero or a one, a period, and then the remainder of
+  the mantissa in hexadecimal form, with digits from 0-9, a-f, or A-F.  The
+  start of the exponent is denoted by a "p", which is then followed by an
+  optional plus or minus sign and then digits from 0 to 9.
```

Comment:
What about this:
Hexadecimal floating-point constant - a hexadecimal constant (starting with "0x prefix) with or without radix separator, followed by "p" and a hexadecimal exponent value (without "0x" prefix). The exponent is never optional for hexadecimal floating-point literals.

Reply:
I made this change. But as you pointed out - starting with 0 or 1 needs some investigation

Reply:
To be clear here - ISPC accepts hexadecimal literals with only 1 or 0 before radix separator, i.e. `0x1.fffp1` or `0x0.fffp1`, which means the fractional part is exactly mantissa, without any modifications. This form is subset of C++17 hexadecimal floating point literals (see [here](https://en.cppreference.com/w/cpp/language/floating_literal) for details), which accept any hexadecimal value and do normalization.

I think it makes sense to accept C++17 literals, especially if/when we switch to APFloat for parsing literals. But his should be another change set. Let's keep current restriction for now.